File: app/src/main/java/com/shakil/barivara/utils/SpinnerData.java
package com.shakil.barivara.utils;

import android.content.Context;
import com.shakil.barivara.R;
import java.util.ArrayList;

public class SpinnerData {

    private final Context context;

    public SpinnerData(Context context) {
        this.context = context;
    }

    public ArrayList<String> setYearData(){
        ArrayList<String> spinnerValues = new ArrayList<>();
        String[] yearArray = {context.getString(R.string.select_data_1),"2019","2020","2021","2022","2023","2024","2025"};
        for (int start = 0; start < yearArray.length; start++) {
            spinnerValues.add(yearArray[start]);
        }
        return spinnerValues;
    }


    public ArrayList<String> setMonthData(){
        ArrayList<String> spinnerValues = new ArrayList<>();
        String[] monthArray = {context.getString(R.string.select_data_1),context.getString(R.string.january),
                context.getString(R.string.february), context.getString(R.string.march), context.getString(R.string.april),
                context.getString(R.string.may), context.getString(R.string.june), context.getString(R.string.july),
                context.getString(R.string.august), context.getString(R.string.september), context.getString(R.string.october),
                context.getString(R.string.november), context.getString(R.string.december)};
        for (int start = 0; start < monthArray.length; start++) {
            spinnerValues.add(monthArray[start]);
        }
        return spinnerValues;
    }

    public ArrayList<String> setTenantTypeData(){
        ArrayList<String> spinnerValues = new ArrayList<>();
        String[] tenantTypeArray = {context.getString(R.string.select_data_1),context.getString(R.string.bachelor),
                context.getString(R.string.family), context.getString(R.string.small_family), context.getString(R.string.female_only),
                context.getString(R.string.male_only)};
        for (int start = 0; start < tenantTypeArray.length; start++) {
            spinnerValues.add(tenantTypeArray[start]);
        }
        return spinnerValues;
    }

    public ArrayList<String> setMeterTypeData(){
        ArrayList<String> spinnerValues = new ArrayList<>();
        String[] meterTypeArray = {"Select Data","Main Meter","Sub Meter"};
        for (int start = 0; start < meterTypeArray.length; start++) {
            spinnerValues.add(meterTypeArray[start]);
        }
        return spinnerValues;
    }

    public ArrayList<String> setSpinnerNoData(){
        ArrayList<String> spinnerValues = new ArrayList<>();
        String[] meterTypeArray = {"No Data"};
        for (int start = 0; start < meterTypeArray.length; start++) {
            spinnerValues.add(meterTypeArray[start]);
        }
        return spinnerValues;
    }
}

File: app/src/main/java/com/shakil/barivara/utils/SpinnerData.kt
package com.shakil.barivara.utils

import android.content.Context
import com.shakil.barivara.R
import java.util.Calendar
import java.util.Collections

class SpinnerData(private val context: Context) {
    fun setYearData(): ArrayList<String> {
        val spinnerValues = ArrayList<String>()
        // Current year
        val currentYear = Calendar.getInstance().get(Calendar.YEAR)
        spinnerValues.add(context.getString(R.string.select_data_1))
        // Add years dynamically up to the current year
        for (year in 2015..currentYear) {
            spinnerValues.add(year.toString())
        }
        return spinnerValues
    }

    fun setMonthData(selectedYear: Int): ArrayList<String> {
        val spinnerValues = ArrayList<String>()
        // Current year and month
        val calendar = Calendar.getInstance()
        val currentYear = calendar.get(Calendar.YEAR)
        val currentMonth = calendar.get(Calendar.MONTH) + 1 // Month is 0-indexed

        spinnerValues.add(context.getString(R.string.select_data_1))

        // Add months dynamically
        for (month in 1..12) {
            // If the selected year is the current year, limit the months to the current month
            if (selectedYear == currentYear && month > currentMonth) {
                break
            }
            // Convert month index to localized month name
            spinnerValues.add(getMonthName(month))
        }

        return spinnerValues
    }

    private fun getMonthName(month: Int): String {
        return when (month) {
            1 -> context.getString(R.string.january)
            2 -> context.getString(R.string.february)
            3 -> context.getString(R.string.march)
            4 -> context.getString(R.string.april)
            5 -> context.getString(R.string.may)
            6 -> context.getString(R.string.june)
            7 -> context.getString(R.string.july)
            8 -> context.getString(R.string.august)
            9 -> context.getString(R.string.september)
            10 -> context.getString(R.string.october)
            11 -> context.getString(R.string.november)
            12 -> context.getString(R.string.december)
            else -> ""
        }
    }

    fun setTenantTypeData(): ArrayList<String> {
        val spinnerValues = ArrayList<String>()
        val tenantTypeArray = arrayOf(
            context.getString(R.string.bachelor),
            context.getString(R.string.large_family),
            context.getString(R.string.small_family),
            context.getString(R.string.female_only),
            context.getString(R.string.male_only),
            context.getString(R.string.others)
        )
        spinnerValues.addAll(tenantTypeArray)
        return spinnerValues
    }

    fun setRentStatusSpinnerData(): ArrayList<String> {
        val spinnerValues = ArrayList<String>()
        val tenantTypeArray = arrayOf(
            context.getString(R.string.paid),
            context.getString(R.string.due),
        )
        spinnerValues.addAll(tenantTypeArray)
        return spinnerValues
    }

    fun setMeterTypeData(): ArrayList<String> {
        val spinnerValues = ArrayList<String>()
        val meterTypeArray = arrayOf("Select Data", "Main Meter", "Sub Meter")
        Collections.addAll(spinnerValues, *meterTypeArray)
        return spinnerValues
    }

    fun setSpinnerNoData(): ArrayList<String> {
        val spinnerValues = ArrayList<String>()
        val meterTypeArray = arrayOf("No Data")
        Collections.addAll(spinnerValues, *meterTypeArray)
        return spinnerValues
    }
}
